zerver/views/auth.py

```diff
@@ -572,6 +572,7 @@ def log_into_subdomain(request: HttpRequest, token: str) -> HttpResponse:
 def redirect_and_log_into_subdomain(result: ExternalAuthResult) -> HttpResponse:
     token = result.store_data()
     realm = get_realm(result.data_dict["subdomain"])
+    assert realm is not None
```

Comment:
Oh, right, in this function we end up still using `get_realm`, not `get_realm_from_request`, so `assert realm is not None` is completely redundant, let's not change anything here after all.